core/src/main/java/hudson/model/AbstractItem.java

```diff
@@ -351,6 +353,11 @@ protected void checkRename(@Nonnull String newName) throws Failure {
      * you can use this method.
      */
     protected void renameTo(final String newName) throws IOException {
+
+        if (!isNameEditable()) {
+            throw new IOException("Trying to rename an item that has not editable name.");
```

Comment:
Not valid English. _Non-editable_ is, but perhaps you meant something like

```suggestion
            throw new IOException("Trying to rename an item that does not support this operation.");
```

?

Reply:
yes thanks it's better, I will take it.